Converted Project to Maven Project + Portal Layout Fixes + General Cleanup

michaeljfazio opened the pull request:
Converted the InvientChart project to a Maven project in the structure prescribed by Vaadin development team.
- Includes assembly for building releases into .zip file (mvn package assembly:assembly)
- Separated demo application from core addon.
- Upgraded javascript libraries (jquery & highcharts).
- Removed unnecessary previous distribution jars.
- Removed all IDE-specific artefacts (use import maven project instead).
- Added .gitignore with target directories and IDE-specific files added to ignore.
- Minor fixes to allow chart drag'n'drop (still has some issues).

INSTRUCTIONS

Compile
- From root module (mvn compile)

Create invient charts jar
- From root module (mvn package)

Create invient charts javadoc
- From root module (mvn javadoc:jar)

Create distributable .zip
- From root module (mvn javadoc:jar package assembly:assembly)

Run in Jetty
- From root module (mvn install), then
- From invient-charts-demo module directory (mvn jetty:run)

michaeljfazio committed:
Converted existing project to Maven project (Vaadin specified structure).
Removed all IDE specific files.
Added gitignore file.
Upgraded demo project javascript sources (highcharts & jquery).
Configured maven assembly for packaging as distributable .zip file.

invient-cp commented:
Hi,
After pulling your sources, I could render all graphs, specifically, "timeseries-zoomable" and "master-detail". 
Can you please let me know if you have already faced this issue?

Thanks

michaeljfazio commented:
Hi,

I fired up the demo and confirmed that I too am seeing what you are seeing. Given that my changes to the actual "meat" of the Java code were relatively minor, I suspected that one of the third party libraries might be at fault. So, I ran with the exact versions of third part libraries from where I forked the repository:
1. GWT 2.1.1
2. Vaadin 6.5.4
3. JQuery 1.4.4
4. Highcharts 2.1.2

Still, the time-series and master-detail charts were not displaying correctly. Furthermore, there is no indication of a client-side JavaScript issue as far as I can tell. Hence, I can only assume one of two things:
1. The minor changes I made to the caching of chart configuration on detach is the issue, or;
2. the bug exists in the code from where I forked.

I can't dedicate much more time at the moment to diagnosing the issue, but I will come back to it as soon as possible. In the meantime, could you confirm the existence of the bug in the version before it was forked in my repository to guarantee that the problem was not inherited?

Thanks

-Michael

invient-cp commented:
Thanks for putting your effort. I built demo from the master repository and ran the demo application. It is working fine. All charts were rendered correctly. The only difference I see is the way widgetset is compiled. I use Vaadin eclipse plugin version-2.0.1 to compile widgetset whereas your changes uses maven vaadin plugin to do the same. 
I am yet to confirm if that is the issue. 

Thanks

axfordl committed:
Ensure chart is resized when parent container is resized

michaeljfazio committed:
Fixed Time Series Zoomable Charts

The issue causing time series zoomable charts to not display was
in-fact caused by the upgrade of JavaScript libraries. It seems as
though my initial attempt at resolving this issue but downgrading said
libraries was thwarted by locally cached JavaScript in the browser.

michaeljfazio commented:
Invient,

After further investigation I discovered that the issue was indeed caused by the JavaScript library upgrade. But, because of locally cached JavaScript, it did not appear to have been fixed. I have now successfully resolved the bug with a script downgrade.

Furthermore, my colleague has included a fix which solves a chart re-sizing issue that previously existed. Both fixes are now available in my repo and can be integrated at your discression.

Sorry for taking such a long time with this fix. I've been very busy!

-Michael

Commit:
Added millisecond format to datetime formats.

michaeljfazio commented:
@invient-cp  we would really love to see this  changeset merged with InvientCharts project. Is there anything else we can do to help make this happen soon? My development team has more bug fixes that we would like to contribute but are holding off until the project has been released as a Maven managed component.

xylo commented:
Any updates on this pull request?  Are there still open issues?  I'm also very interested in a proper maven support since I'm not using eclipse.  Hope it's getting merged soon.

michaeljfazio commented:
Xylo. It would appear as though this project is stagnating somewhat as the original authors have gone a tad quiet. I suspect that they are quite busy with other things and simply cannot dedicate the time to continued development (fair enough).

If this pull request is still not actioned in the new year I will re-release the project under a different name in order to try and spark re-newed community development. Hopefully I can find some suitable developers to also pick up some of the administration tasks from the Vaadin forums.

Furthermore, you might be pleased to know that the Vaadin team are apparently developing their own Highcharts library as a paid-for professional add-on in conjunction with the highcharts/stockcharts library developers. As such this project could become completely redundant in the near future (provided you are willing to pay for the add-on).

-Michael

xylo commented:
I agree.  Before the development gets completely stuck it's better to re-release this add-on under a different name.  Thanks for your efforts.
Moreover I would regret a removal of this add-on from the vaadin directory in favor of a commercial add-on, since I'm using this add-on for non-commercial purposes.

michaeljfazio commented:
Not to worry xylo, I was not suggesting that the InvientCharts addon would be removed from the addon directory. Watch this space with regards to releasing InvientCharts under a new project name.

Hopefully we can slot in some improvements and unit tests as well ;)

SakulK committed:
added a few missing options

SakulK committed:
borderWidth in pie config changed from int to double

michaeljfazio committed:
Merge pull request #2 from SakulK/master

- Added ignore hidden point in series option.
- Added support for HTML in tool tip options.
- Added header and point format.
- Changed border width to be type double rather than int.

michaeljfazio committed:
Added SCM details to project POM.

michaeljfazio committed:
Fixed missing SCM URL prefix.

michaeljfazio committed:
[maven-release-plugin] prepare release invient-charts-1.0

michaeljfazio committed:
Iterating development version to 1.1-SNAPSHOT

SakulK committed:
changed the crosshairs property of tooltip from a simple boolean to an object to support some options available in highcharts (this is support only for the X-axis crosshair)

SakulK committed:
remove log message

SakulK committed:
fixed itemStyle in Legend

mletenay committed:
Fix rendering several charts in one component.

Use counter for widget ids, currentTimeMillis can have low granularity
and assign same ID to several widgets

SakulK committed:
fixed performance issues in DateTimePoint and added turboThreshold option to plotOptions

michaeljfazio committed:
Merge https://github.com/SakulK/InvientCharts

Conflicts:
	invient-charts-demo/pom.xml
	invient-charts/pom.xml
	pom.xml

michaeljfazio committed:
Merge https://github.com/mlaitinen/InvientCharts

Conflicts:
	invient-charts/src/main/java/com/invient/vaadin/charts/InvientChartsConfig.java
	invient-charts/src/main/java/com/invient/vaadin/charts/InvientChartsUtil.java
	invient-charts/src/main/java/com/invient/vaadin/charts/widgetset/client/ui/VInvientCharts.java

michaeljfazio committed:
Specified maven jetty plugin version.

michaeljfazio committed:
Minor modifications to com.invient.vaadin.charts.Color javadoc.

michaeljfazio committed:
Minor cleanup in to com.invient.vaadin.charts.Gradient

michaeljfazio committed:
Minor changes to com.invient.vaadin.charts.InvientChartsConfig javadoc

michaeljfazio committed:
Multiple updates to javadoc.
Added generics support for Point classes.
